### ISSUE TITLE
Tab Manager Plus stopped working, Options and Manager are empty

### ISSUE THREAD

**Issue by JashanChittesh** [stefanXO/Tab-Manager-Plus#209]
Tab Manager Plus worked just fine for me until a few days ago. My current Firefox version is 113.0.1.

Now, when I click on the tabs view to select a tab, that view comes up empty. Also, when I enter the Options in the Extension settings, they briefly flash up but then immediately disappear. In the Browser Console, I get an error after trying to open the tabs view: 

```
WebExtension context not found! [ExtensionParent.jsm:1296](resource://gre/modules/ExtensionParent.jsm)
    getContextById resource://gre/modules/ExtensionParent.jsm:1296
    recvAPICall resource://gre/modules/ExtensionParent.jsm:1133
    _recv resource://gre/modules/ConduitsChild.jsm:84
    receiveMessage resource://gre/modules/ConduitsParent.jsm:466
```

Here's what I see: 

![TabManagerPlusForFirefoxBroken](https://github.com/stefanXO/Tab-Manager-Plus/assets/986684/f21b9a52-5dae-47c7-8338-a229a1e19a7f)

EDIT: I have also tried downgrading to Firefox 112.0.2 but that didn't fix the issue (and instead caused a whole lot of other issues). And the plugin still works fine on my laptop (which also has 113.0.1 installed but is on Windows 11 while my desktop is on Windows 10).

**Comment by travisbhartwell** [comment]
I also am experiencing this recently.

It doesn't seem tied to the Firefox version, as I was able to create a new profile in Firefox and install the extension in the new profile and it works there. This is not an ideal fix for me, as I don't want to lose my history, bookmarks, and other configuration in my primary profile.

But it feels like there must be a configuration or some setting that has got messed up in my existing profile that is causing the extension to stop working.

**Comment by decard123** [comment]
similar problem
just happened today
latest FF. win10 x64
no errors: just a blank form
have turned off/on. have uininstalled and reinstalled - same prob
i figure it is a TMP bug fix; but maybe a FF chnage that has disabled this app.

**Comment by decard123** [comment]
TMP just started working again. 

not sure what i did: i was playing around trying some other tab manager extensions, and, poof, TMP came up

**Comment by username930** [comment]
The URL on some tabs is not defined. I was using it with like. suspended tabs. that don't load sometimes. and maybe this causes tab URL to be undefined. the fix is to keep closing tabs until it works. maybe you have to find and close crashed suspended tabs.

Uncaught TypeError: this.refs.root is undefined
TabManager.js:560:6

TypeError: hostname is undefined
moz-extension://###/outlib/Window.js:364
[react-dom.production.min.js:112:381](moz-extension://###/vendor/react-dom.production.min.js)


var hostname = url.hostname;
									if (protocol.indexOf("chrome-extension") > -1) {
										hostname = tabs[i].props.tab.title;
									} else if (protocol.indexOf("about") > -1) {
										hostname = tabs[i].props.tab.title;
									} else if (hostname.indexOf("mail.google") > -1) {
										hostname = "gmail";
									} else {
										hostname = hostname.replace("www.", "");
										var regex_var = new RegExp(/(\.[^\.]{0,2})(\.[^\.]{0,2})(\.*$)|(\.[^\.]*)(\.*$)/);
										hostname = hostname.
										replace(regex_var, "").
										split(".").
										pop();
									}
									if (hostname.length > 18) {

**Comment by username930** [comment]
The URL on some tabs is not defined. I was using it with like. suspended tabs. that don't load sometimes. and maybe this causes tab URL to be undefined. the fix is to keep closing tabs until it works. maybe you have to find and close crashed suspended tabs.

Uncaught TypeError: this.refs.root is undefined
TabManager.js:560:6

TypeError: hostname is undefined
moz-extension://###/outlib/Window.js:364
react-dom.production.min.js:112:381

var hostname = url.hostname;
if (protocol.indexOf("chrome-extension") > -1) {
hostname = tabs[i].props.tab.title;
} else if (protocol.indexOf("about") > -1) {
hostname = tabs[i].props.tab.title;
} else if (hostname.indexOf("mail.google") > -1) {
hostname = "gmail";
} else {
hostname = hostname.replace("www.", "");
var regex_var = new RegExp(/(.[^\.]{0,2})(.[^\.]{0,2})(.$)|(.[^\.])(.*$)/);
hostname = hostname.
replace(regex_var, "").
split(".").
pop();
}
if (hostname.length > 18) {

**Comment by max-rh** [comment]
How do i use this new Extension that you forked ? I tried to load it manually, but its giving me that the file is corrupt

**Comment by hatemicroshit** [comment]
username930: as per your suggestion, i used [a session manager](https://addons.mozilla.org/en-US/firefox/addon/tab-session-manager/) to save the current session and replace with the identical saved session, which reloaded 500 tabs in an instant without crashing the browser. saved favicons are gone but it works again
![image_2024-02-15_12-20-28](https://github.com/stefanXO/Tab-Manager-Plus/assets/119356180/0986b40a-1231-4e8c-b5f9-e517cdbea8e7)

**Comment by mastef** [comment]
Can you try the latest firefox add-on version if it solves the issue for you?
https://addons.mozilla.org/en-US/firefox/addon/tab-manager-plus-for-firefox/

**Comment by mastef** [comment]
Should be fixed in 5dc30eda96037ea9626821eaa4d5fcd362d655f5 / 5.2.1
